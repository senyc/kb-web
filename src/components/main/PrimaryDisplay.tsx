import { TextEntry } from '@annotations';

import Distribution from './text/Distribution';
import Editor from './text/Editor';
import InternshipOne from './text/InternshipOne';
import InternshipTwo from './text/InternshipTwo';
import Languages from './text/Languages';
import Welcome from './text/Welcome';
import Typetest from './text/Typetest';
import KBWeb from './text/KBWeb';

interface PrimaryDisplayProps {
  textDisplay: TextEntry;
}

export default function PrimaryDisplay({ textDisplay }: PrimaryDisplayProps) {
  return (
    <div className='mb-11 ml-11 mt-32 basis-7/12 self-center sm:mb-0 sm:mt-64 md:ml-32 xl:mt-0'>
      {
        (() => {
          switch (textDisplay) {
            case TextEntry.Welcome:
              return <Welcome />;
            case TextEntry.Distro:
              return <Distribution />;
            case TextEntry.Editor:
              return <Editor />;
            case TextEntry.Languages:
              return <Languages />;
            case TextEntry.InternshipOne:
              return <InternshipOne />;
            case TextEntry.InternshipTwo:
              return <InternshipTwo />;
            case TextEntry.Typetest:
              return <Typetest />;
            case TextEntry.Site:
              return <KBWeb />;
            default:
              return <Welcome />;
          }
        })()
      }
    </div>
  );
}
