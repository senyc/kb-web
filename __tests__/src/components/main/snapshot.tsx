import { create } from 'react-test-renderer';

import { TextEntry } from '@annotations';

import StyledLine from '@components/main/ColoredLine';
import PrimaryDisplay from '@components/main/PrimaryDisplay';

describe('main elements match recorded snapshots', () => {
  describe('StyledLine', () => {
    it('renders on the bottom', () => {
      const tree = create(<StyledLine displayOn='left' />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders on the left', () => {
      const tree = create(<StyledLine displayOn='bottom' />).toJSON();
      expect(tree).toMatchSnapshot();
    });

  });

  describe('Primary Display', () => {
    it('displays welcome correctly', () => {
      const tree = create(<PrimaryDisplay textDisplay={TextEntry.Welcome}/>).toJSON();
      expect(tree).toMatchSnapshot();
    });

  });
});
