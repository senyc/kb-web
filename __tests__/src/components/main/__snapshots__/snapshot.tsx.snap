// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main elements match recorded snapshots Primary Display displays welcome correctly 1`] = `
<div
  className="mb-11 ml-11 mt-32 basis-7/12 self-center sm:mb-0 sm:mt-64 md:ml-32 xl:mt-0"
>
  <h2
    className="mb-5 text-4xl font-bold sm:text-5xl"
  >
    Kyler Bomhof
  </h2>
  <div
    className="relative ml-5 mr-5 max-w-xl font-medium"
  >
    <p
      className="mb-5 font-medium "
    >
      Hi, I'm a passionate full stack developer with professional experience building web applications.
    </p>
    <p
      className="mb-5 font-medium"
    >
      I'm passionate about software architecture and building applications such that they are testable, and actually tested.
    </p>
    <span
      className="styled-line absolute -left-5 bottom-0 top-0 w-0.5"
    />
  </div>
</div>
`;

exports[`main elements match recorded snapshots StyledLine renders on the bottom 1`] = `
<span
  className="styled-line absolute -left-5 bottom-0 top-0 w-0.5"
/>
`;

exports[`main elements match recorded snapshots StyledLine renders on the left 1`] = `
<span
  className="styled-vertical-line absolute -bottom-1 left-0 h-px px-11"
/>
`;
